src/Build/BackEnd/Components/SdkResolution/NuGet/NuGetSdkResolver.cs

```
@@ -0,0 +1,202 @@
+                        IReadOnlyList<RestoreResultPair> results = RestoreRunnerEx.RunWithoutCommit(
+                                context.ProjectFilePath,
+                                sdk.Name,
+                                parsedSdkVersion.ToFullString(),
+                                settings,
+                                new NuGetSdkLogger(context.Logger, warnings, errors))
+                            .Result;
+
+                        // Look for a successful result, any errors are logged by NuGet
+                        foreach (RestoreResult result in results.Select(i => i.Result).Where(i => i.Success))
+                        {
+                            // Find the information about the package that was installed.  In some cases, the version can be different than what was specified (like you specify 1.0 but get 1.0.0)
+                            LibraryIdentity installedPackage = result.GetAllInstalled().FirstOrDefault(i => i.Name.Equals(sdk.Name));
+
+                            if (installedPackage != null)
+                            {
+                                if (!TryGetMSBuildSdkPackageInfo(fallbackPackagePathResolver, installedPackage.Name, installedPackage.Version, out installedPath, out installedVersion))
+                                {
+                                    // This should never happen
+                                    errors.Add("Package was installed but could not be found");
+                                }
+                            }
+                            else
+                            {
+                                // This should never happen
+                                errors.Add("Package was supposed to be installed but wasn't");
+                            }
+                        }
+                    }
+                    catch (Exception e)
```

This is a larger scope than I'm really comfortable with for a catch-all exception handler. Is there no NuGet exception base class?